python/cudf/cudf/core/column/decimal.py

```diff
@@ -55,11 +55,27 @@ def to_arrow(self):
     def binary_operator(self, op, other, reflect=False):
         if reflect:
             self, other = other, self
-        result = libcudf.binaryop.binaryop(self, other, op, "int32")
+        scale = _binop_scale(self.dtype, other.dtype, op)
+        output_type = Decimal64Dtype(
+            scale=scale, precision=18
```

Comment:
Would it make sense for the `Decimal64Dtype` constructor to default `precision` to 18?

Reply:
So we can avoid the magic number `18` all over our code?

Reply:
It helps for hooking into libcudf APIS, so as long as it doesn't cause issues in the Python code - sounds like a decent idea to me :)

Reply:
@kkraus14 any objections? `18` being the max precision the type can support. Note that this will deviate from PyArrow which requires the user to specify a precision explicitly.

Reply:
I made the changes locally and it cleans up the code nicely. Will wait on @kkraus14 before I push

Reply:
That feels like a bit of a slippery slope to me. Since `scale` defaults to `0`, that would allow someone to do `cudf.Decimal64Dtype()` and it silently gives them something instead of guiding them to specify the precision.

Reply:
But it also kinda sucks having to specify `precision=18` everywhere. How about a class attribute `Decimal64Dtype.MAX_PRECISION` instead?

Reply:
I like the `MAX_PRECISION` attribute, but do we want it to be internal or public?

Reply:
I think it's appropriate to make it public

Reply:
@codereport can we make `MAX_PRECISION` a class attribute and use that here instead?

In `dtypes.py`:

```python
class DecimalDtype:
    MAX_PRECISION = 18
```

Here:

```suggestion
            scale=scale, precision=cudf.Decimal64Dtype.MAX_PRECISION
```

Reply:
Pushed the changes. Turns out we already had `MAX_PRECISION` :facepalm: 
```py
MAX_PRECISION = np.floor(np.log10(np.iinfo("int64").max))
```

Reply:
:facepalm: sry, I changed 1 occurrence of 18, missed the 2nd. pushed now.